lib/sparql_to_sw_solr/instance_title_fields.rb

```diff
@@ -8,14 +8,22 @@ def add_doc_title_fields(doc)
         # TODO: what if primary title values aren't single values?
         primary_solutions = primary_title_result
         primary_main_title = values_from_solutions(primary_solutions, 'mainTitle').first
+
         doc[:title_245a_search] = primary_main_title
         doc[:title_245_search] = values_from_solutions(primary_solutions, 'rdf-schema#label').first
         doc[:title_245a_display] = primary_main_title.strip.gsub(/[\\,:;\/ ]+$/, '') if primary_main_title
+
         primary_subtitle = values_from_solutions(primary_solutions, 'subtitle').first
+
         # TODO: what should separator be? what if main title ends with '=' ?
-        doc[:title_display] = "#{primary_main_title}, #{primary_subtitle}"
+        doc[:title_display] = "#{primary_main_title.strip.gsub(/[\\,:;\/ ]+$/, '') if primary_main_title}" \
+                              "#{primary_subtitle.present? ? ' : ' : nil}" \
+                              "#{primary_subtitle.strip.gsub(/[\\,:;\/ ]+$/, '') if primary_subtitle}"
+
         resp_statement = values_from_solutions(primary_solutions, 'responsibilityStatement').first
-        doc[:title_full_display] = "#{doc[:title_display]} / #{resp_statement}"
+
+        doc[:title_full_display] = "#{doc[:title_display]}#{resp_statement.present? ? ' / ' : nil}" \
+                                   "#{resp_statement.gsub(/[\\,:;\/ ]+$/, '') if resp_statement}"
```

Comment:
ditto to comment for title_display code ... if you think it's clearer with my suggestion, please reapply it here.